### PR TITLE
Added unit tests for ReadLogs

### PR DESCRIPTION
**Pull request by brianpursley**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
`ReadLogs()` is missing unit tests. This PR adds unit tests.

**Which issue(s) this PR fixes**:
Adds unit tests which will help in fixing issue #95089 in another PR (https://github.com/kubernetes/kubernetes/pull/93920)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

### CODE REVIEW

**Review by RobertKielty — CHANGES_REQUESTED** [review]
Thanks for writing these unit tests! 

I added some comments on a test name and suggested adding another TailLines test for line numbers great than the file contents.

The reason I suggested not extracting the 2 as the variable was I had to go looking for it when I was reading the testcase and then when I did locate it the variable name did not hint at the value 2 in any way.

**Comment by brianpursley** [comment]
@RobertKielty thanks I will take a look at the suggestions.

Regarding extracting some of the numbers into variables, the properties in podLogOptions are *int, so I didn't know how to do that without introducing a variable.

File: pkg/kubelet/kuberuntime/logs/logs_test.go [hunk 1]
```diff
@@ -66,6 +70,139 @@ func TestLogOptions(t *testing.T) {
...
+		podLogOptions v1.PodLogOptions
+		expected      string
+	}{
+		{
+			name:          "default pod log options should output all lines",
+			podLogOptions: v1.PodLogOptions{},
+			expected:      "line1\nline2\nline3\n",
+		},
+		{
+			name: "using TailLines should output only last N lines",
```
[the remark this answers is not in the document]
Comment (brianpursley) [reply pkg/kubelet/kuberuntime/logs/logs_test.go]: Fixed and added tests for when `TailLines` and `LimitBytes` exceed the log size

File: pkg/kubelet/kuberuntime/logs/logs_test.go [hunk 2]
```diff
@@ -66,6 +70,139 @@ func TestLogOptions(t *testing.T) {
...
+		},
+		{
+			name: "using follow should output all log lines",
+			podLogOptions: v1.PodLogOptions{
+				Follow: true,
+			},
+			expected: "line1\nline2\nline3\n",
+		},
+		{
+			name: "using follow combined with TailLines should output the N most recent log lines",
```
[the remark this answers is not in the document]
Comment (brianpursley) [reply pkg/kubelet/kuberuntime/logs/logs_test.go]: Fixed

**Comment by brianpursley** [comment]
@RobertKielty ptal

File: pkg/kubelet/kuberuntime/logs/logs_test.go [hunk 3]
```diff
@@ -18,6 +18,11 @@ package logs
 
 import (
 	"bytes"
+	"context"
+	"io/ioutil"
+	testing2 "k8s.io/cri-api/pkg/apis/testing"
```
Comment (zhouya0) [inline pkg/kubelet/kuberuntime/logs/logs_test.go:23]: Name `testing2`,  sounds `apitest` is better?

**Review by zhouya0 — COMMENTED** [review]
lgtm and thanks !

File: pkg/kubelet/kuberuntime/logs/logs_test.go (hunk 3 above)
Comment (brianpursley) [reply pkg/kubelet/kuberuntime/logs/logs_test.go]: good catch. Goland chose that name for me and I didn't notice. I renamed it just now.

**Review by RobertKielty — APPROVED** [review]

**Pull request merged**